torch/_dynamo/eval_frame.py

```diff
@@ -793,41 +1010,11 @@ def result_capturing_wrapper(*graph_inputs):
     flat_both = list(graph_captured_result) + flat_args
     matched_output_elements_positions = produce_matching(flat_both, flat_results_traced)
 
-    class ChangeInputOutputSignature(torch.fx.interpreter.Transformer):
-        def __init__(
-            self,
-            m,
-        ):
-            super().__init__(m)
-            arg_len = len(flat_args)
-            self.new_args = [
-                super(ChangeInputOutputSignature, self).placeholder(f"arg{i}", (), {})
-                for i in range(0, arg_len)
-            ]
-            self.old_args_gen = (
-                self.new_args[i] for i in matched_input_elements_positions
-            )
-
-        def placeholder(self, target, args, kwargs):
-            arg = next(self.old_args_gen)
-            if "val" in self.current_node.meta:
-                arg.node.meta["val"] = self.current_node.meta["val"]
-            if "tensor_dict" in self.current_node.meta:
-                arg.node.meta["tensor_dict"] = self.current_node.meta["tensor_dict"]
-            return arg
-
-        def output(self, target, args, kwargs):
-            dynamo_result_flat = args[0]
-            lookup = [*dynamo_result_flat, *self.new_args]
-            new_result_flat = [lookup[i] for i in matched_output_elements_positions]
-            return super().output(target, (new_result_flat,), {})
-
-        def run_node(self, n):
-            self.current_node = n
-            r = super().run_node(n)
-            if "val" in self.current_node.meta:
-                r.node.meta["val"] = self.current_node.meta["val"]
-            return r
+    graph = _AddRuntimeAssertsInInputConstraint(
```

Comment:
I don't understand why we want this as a default? Like, this feels like something that should exist entirely outside of dynamo - anyone is free to add any passes or transforms they want, but this is far too assumptive of a single usecase.

Reply:
We currently don't have agreed final export API yet. So there is no other place to put this pass at the moment. I can move it out of dynamo once we have that API. Are you ok with it if I hide it under a flag?

Reply:
What I would like to discuss is the direction of passes like this in dynamo. To me, this feels like a laborious and assumptive default, or inclusion. If someone needs this, they can invoke a util we provide. 

w/r/t hiding it under a flag - export growing in flags and flag combinations is not a good direction. 

The burden of proof of including a new feature is that it is generally useable, and I think promotion of guards to asserts does not cross that bar. Offering a util that does it. and putting the burden on the .export caller to use it seems sufficient.

Reply:
Thanks for the comment @voznesenskym. Tugsuu and I analyzed the different options and we also feel that it is better to implement as a util that can be invoked after dynamo export call, so that dynamo can maintain the contract of "making assumptions, leave it to the caller as to how to enforce them"